File: examples/python/table_column.py
import html as _html


class TableColumn:
    def __init__(self, index=0, title=""):
        self.index = index
        self.title = title
        self.data_type = "text"
        self.width = 100
        self.sortable = False
        self.visible = True

    def set_title(self, t):
        self.title = t

    def set_width(self, w):
        self.width = w

    def render_header(self):
        if not self.visible:
            return ""
        style = f' style="width:{self.width}px"' if self.width else ""
        return f"<th{style}>{_html.escape(self.title)}</th>"

    def render_cell(self, value):
        if not self.visible:
            return ""
        return f"<td>{_html.escape(str(value))}</td>"

    def sort(self, rows):
        return sorted(rows, key=lambda r: r.get_cell(self.index).value)

    def hide(self):
        self.visible = False

    def show(self):
        self.visible = True

File: examples/c++/table_column.cpp
#include <string>
#include <vector>
#include <algorithm>

class TableColumn {
public:
    int index = 0;
    std::string title;
    std::string dataType = "text";
    int width = 100;
    bool sortable = false;
    bool visible = true;
    void setTitle(const std::string& t) { title = t; }
    void setWidth(int w) { width = w; }
    void sort(std::vector<TableRow>& rows) {
        std::sort(rows.begin(), rows.end(), [this](const TableRow& a, const TableRow& b) {
            return a.getCell(index).value < b.getCell(index).value;
        });
    }
    void hide() { visible = false; }
    void show() { visible = true; }
};
